### DISCUSSION TITLE
Demo not working on esp32-2432S028R

### DISCUSSION THREAD

**Discussion by pelizz** [rzeldent/esp32-smartdisplay#92] [Q&A] [answered]
Hello,
i tried the demo app (https://github.com/rzeldent/esp32-smartdisplay-demo) on esp32-2432S028R and i got this error and the program show blank screen:
E (450) spi_master: check_trans_valid(689): txdata transfer > host maximum
E (451) lcd_panel.io.spi: panel_io_spi_tx_color(380): spi transmit (queue) color failed.

Some indications to solve the problem?
Thank you.

**Comment by rzeldent** [comment, marked as answer]
Hi pelizz,

Did you updated all, including the boards directory submodules?
Please try a
```
git clone --recurse-submodules https://github.com/rzeldent/esp32-smartdisplay-demo.git
```

**Comment by pelizz** [comment]
Hi...
thanks for the answer...yes i have done the clone with submodules as documented...but i have the same error.
I do not use a Sunton board ma a generic esp32-yellowboard 2.8 inch.

**Reply by rzeldent** [reply]
Use the develop version. Just tested and works on my esp32-2432S028R
You have to enable the board for your version. So enable (remove #) before the esp32-2432S028R

**Comment by chrbcd** [comment]
Hello,
I have the same problem. RGB ok, sound ok but screen white.

**Comment by rzeldent** [comment]
Can you try the develop branch (and also the boards submodule in develop)?
Probably something already fixed.

**Comment by chrbcd** [comment]
Not completed. But I think VSCode install master version every time.

**Reply by rzeldent** [reply]
I'm in the process of merging. So hopefully in a few hours will all be on main

**Comment by chrbcd** [comment]
Hello,
I tested with the new version, still the same problem with an ESP32-2432S028 card with envs ESP32-2432S08R and also v2. RGB LED ok, sound ok, touch ok (logs) but screen white. The card works with a test program (TFT_Rainbow) with TFT-espi (except touch and RGB inversion - BGR).
Also tested with ESP32-1732S09N and ESP32-4827S043 cards, everything is ok.

Some indications to solve the problem?
Thank you.

**Reply by rzeldent** [reply]
Hi Thanks for testing!

I checked this before and was working on my esp32-2432S028. 

If you have a serial connection and you do a reset do you see any warning/error messages?
Does this fix to set the size helps?

**Comment by chrbcd** [comment]
No errors messages
Serials messages : 

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13232
load:0x40080400,len:3028
entry 0x400805e4
[    25][D][esp32-hal-cpu.c:244] setCpuFrequencyMhz(): PLL: 480 / 2 = 240 Mhz, APB: 80000000 Hz
[   297][I][main.cpp:26] setup(): CPU: ESP32-D0WD-V3 rev3, CPU Freq: 240 Mhz, 2 core(s)
[   298][I][main.cpp:27] setup(): Free heap: 250148 bytes
[   300][I][main.cpp:28] setup(): Free PSRAM: 0 bytes
[   305][I][main.cpp:29] setup(): SDK version: v4.4.6-dirty
[   328][D][WiFiGeneric.cpp:1039] _eventCallback(): Arduino Event: 0 - WIFI_READY
[   406][V][WiFiGeneric.cpp:97] set_esp_interface_ip(): Configuring Station static IP: 0.0.0.0, MASK: 0.0.0.0, GW: 0.0.0.0
[   404][V][WiFiGeneric.cpp:340] _arduino_event_cb(): STA Started
[   411][D][WiFiGeneric.cpp:1039] _eventCallback(): Arduino Event: 2 - STA_START
[   466][V][WiFiGeneric.cpp:355] _arduino_event_cb(): STA Connected: SSID: ChrBcdRes, BSSID: 70:fc:8f:3e:02:62, Channel: 11, Auth: WPA2_PSK
[   468][D][WiFiGeneric.cpp:1039] _eventCallback(): Arduino Event: 4 - STA_CONNECTED
[  2676][V][WiFiGeneric.cpp:369] _arduino_event_cb(): STA Got New IP:192.168.0.4
[  2677][D][WiFiGeneric.cpp:1039] _eventCallback(): Arduino Event: 7 - STA_GOT_IP
[  2680][D][WiFiGeneric.cpp:1102] _eventCallback(): STA IP: 192.168.0.4, MASK: 255.255.255.0, GW: 192.168.0.254
[  2726][V][ssl_client.cpp:321] stop_ssl_socket(): Cleaning SSL connection.
[  2768][D][esp32_smartdisplay.c:209] smartdisplay_init(): smartdisplay_init
[  2771][D][lvgl_ili9341_spi.c:33] lvgl_lcd_init(): lvgl_lcd_init
[  3250][D][esp32_smartdisplay.c:109] smartdisplay_lcd_set_backlight(): smartdisplay_lcd_set_backlight. duty:0
[  3251][D][lvgl_xpt2046_spi.c:34] lvgl_touch_init(): lvgl_touch_init
[  3258][D][esp32_smartdisplay.c:36] lvgl_update_callback(): lvgl_update_callback
[  3282][D][esp32_smartdisplay.c:162] smartdisplay_led_set_rgb(): smartdisplay_led_set_rgb. R:1, G:0, B:0
[  3283][D][esp32-hal-adc.c:194] __analogReadMilliVolts(): eFuse Vref: Supported
[  3289][I][esp32-hal-adc.c:235] __analogReadMilliVolts(): ADC1: Characterized using eFuse Vref: 1093

[  3461][V][ssl_client.cpp:321] stop_ssl_socket(): Cleaning SSL connection.
[  3550][V][ssl_client.cpp:321] stop_ssl_socket(): Cleaning SSL connection.
[  3794][D][esp32_smartdisplay.c:162] smartdisplay_led_set_rgb(): smartdisplay_led_set_rgb. R:1, G:0, B:0
[  4297][D][esp32_smartdisplay.c:162] smartdisplay_led_set_rgb(): smartdisplay_led_set_rgb. R:0, G:0, B:0
[  4800][D][esp32_smartdisplay.c:162] smartdisplay_led_set_rgb(): smartdisplay_led_set_rgb. R:0, G:0, B:0
[  5301][D][esp32_smartdisplay.c:162] smartdisplay_led_set_rgb(): smartdisplay_led_set_rgb. R:0, G:0, B:0
[  5802][D][esp32_smartdisplay.c:162] smartdisplay_led_set_rgb(): smartdisplay_led_set_rgb. R:0, G:0, B:0
[  6308][D][esp32_smartdisplay.c:162] smartdisplay_led_set_rgb(): smartdisplay_led_set_rgb. R:1, G:0, B:0
[  6809][D][esp32_smartdisplay.c:162] smartdisplay_led_set_rgb(): smartdisplay_led_set_rgb. R:1, G:0, B:0
[  7310][D][esp32_smartdisplay.c:162] smartdisplay_led_set_rgb(): smartdisplay_led_set_rgb. R:1, G:0, B:0
[  7811][D][esp32_smartdisplay.c:162] smartdisplay_led_set_rgb(): smartdisplay_led_set_rgb. R:1, G:0, B:0
[  8312][D][esp32_smartdisplay.c:162] smartdisplay_led_set_rgb(): smartdisplay_led_set_rgb. R:0, G:1, B:1
[  8316][D][lvgl_xpt2046_spi.c:26] xpt2046_lvgl_touch_cb(): Pressed at: (142,234), strength: 1073422654
[  8355][D][lvgl_xpt2046_spi.c:26] xpt2046_lvgl_touch_cb(): Pressed at: (145,233), strength: 1073422654

**Comment by rzeldent** [comment]
No error messages at all. Only minor thing I noticed is the %d instead of a %f for the set backlight.
And the esp32-2432S028Rv2? Identical?

**Comment by chrbcd** [comment]
yes identical

**Comment by rzeldent** [comment]
I checked the logging: identical to mine. I added an extra log line to log if a default initialization is present. This is available in the **develop** version of the demo and smartdisplay library. See output below.
I'd like to be sure this special initialization is present when you use the esp32-2432S028Rv2

Could you give it a spin?

Looks identical to https://github.com/rzeldent/esp32-smartdisplay-demo/issues/16


[  2927][D][esp32_smartdisplay.c:209] smartdisplay_init(): smartdisplay_init
[  2929][D][lvgl_ili9341_spi.c:33] lvgl_lcd_init(): lvgl_lcd_init
**[  2931][D][lvgl_ili9341_spi.c:75] lvgl_lcd_init(): Initialization with vendor config**
[  3412][D][esp32_smartdisplay.c:109] smartdisplay_lcd_set_backlight(): smartdisplay_lcd_set_backlight

**Comment by rzeldent** [comment]
Maybe manually setting the vendor config might help find the problem. This is the alternate driver initialzation sequence mentioned in https://github.com/rzeldent/esp32-smartdisplay-demo/issues/16

      "'-D ILI9341_DEV_CONFIG_VENDOR_CONFIG=\"(ili9341_vendor_config_t[]){{.init_cmds=(ili9341_lcd_init_cmd_t[]){{.cmd=0xCF,.data=(uint8_t[]){0x00,0xC1,0x30},.data_bytes=3},{.cmd=0xED,.data=(uint8_t[]){0x64,0x03,0x12,0x81},.data_bytes=4},{.cmd=0xE8,.data=(uint8_t[]){0x85,0x00,0x78},.data_bytes=3},{.cmd=0xCB,.data=(uint8_t[]){0x39,0x2C,0x00,0x34,0x02},.data_bytes=5},{.cmd=0xF7,.data=(uint8_t[]){0x20},.data_bytes=1},{.cmd=0xEA,.data=(uint8_t[]){0x00,0x00},.data_bytes=2},{.cmd=0xC0,.data=(uint8_t[]){0x10},.data_bytes=1},{.cmd=0xC1,.data=(uint8_t[]){0x00},.data_bytes=1},{.cmd=0xC5,.data=(uint8_t[]){0x30,0x30},.data_bytes=2,},{.cmd=0xC7,.data=(uint8_t[]){0xB7},.data_bytes=1},{.cmd=0x3A,.data=(uint8_t[]){0x55},.data_bytes=1},{.cmd=0x36,.data=(uint8_t[]){0x08},.data_bytes=1},{.cmd=0xB1,.data=(uint8_t[]){0x00,0x1a},.data_bytes=2},{.cmd=0xB6,.data=(uint8_t[]){0x08,0x82,0x27},.data_bytes=3},{.cmd=0xF2,.data=(uint8_t[]){0x00},.data_bytes=1},{.cmd=0x26,.data=(uint8_t[]){0x01},.data_bytes=1},{.cmd=0xE0,.data=(uint8_t[]){0x0F,0x2A,0x28,0x08,0x0E,0x08,0x54,0xA9,0x43,0x0A,0x0F,0x00,0x00,0x00,0x00},.data_bytes=15},{.cmd=0xE1,.data=(uint8_t[]){0x00,0x15,0x17,0x07,0x11,0x06,0x2B,0x56,0x3C,0x05,0x10,0x0F,0x3F,0x3F,0x0F},.data_bytes=15},{.cmd=0x2B,.data=(uint8_t[]){0x00,0x00,0x01,0x3F},.data_bytes=4},{.cmd=0x2A,.data=(uint8_t[]){0x00,0x00,0x00,0xEF},.data_bytes=4},{.cmd=0x21},{.cmd=0x11,.delay_ms=120},{.cmd=0x29,.delay_ms=1}},.init_cmds_size=23}}\"'",

**Comment by chrbcd** [comment]
Hello,
With the requested changes, I get the following message when compiling:

.pio/libdeps/esp32-2432S028Rv2/esp32_smartdisplay/src/lvgl_ili9341_spi.c: In function 'lvgl_lcd_init':
<command-line>: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
<command-line>: note: in definition of macro 'ILI9341_DEV_CONFIG_VENDOR_CONFIG'

The resulting log is:

[  3072][D][WiFiGeneric.cpp:1102] _eventCallback(): STA IP: 192.168.0.4, MASK: 255.255.255.0, GW: 192.168.0.254
[  3128][V][ssl_client.cpp:321] stop_ssl_socket(): Cleaning SSL connection.
[  3207][D][esp32_smartdisplay.c:209] smartdisplay_init(): smartdisplay_init
[  3209][D][lvgl_ili9341_spi.c:33] lvgl_lcd_init(): lvgl_lcd_init
[  3211][D][lvgl_ili9341_spi.c:75] lvgl_lcd_init(): Initialization with vendor config
[  3693][D][esp32_smartdisplay.c:109] smartdisplay_lcd_set_backlight(): smartdisplay_lcd_set_backlight. duty:0.500000
[  3695][D][lvgl_xpt2046_spi.c:34] lvgl_touch_init(): lvgl_touch_init
[  3702][D][esp32_smartdisplay.c:36] lvgl_update_callback(): lvgl_update_callback
[  3727][D][esp32_smartdisplay.c:162] smartdisplay_led_set_rgb(): smartdisplay_led_set_rgb. R:1, G:0, B:0

For your information, I added a log line:

	if (panel_dev_config.vendor_config)
        log_d("Initialization with vendor config");
    log_d("Vendor config value %s\n", panel_dev_config.vendor_config); 
    esp_lcd_panel_handle_t panel_handle;

Which give:

[  3303][D][esp32_smartdisplay.c:209] smartdisplay_init(): smartdisplay_init
[  3306][D][lvgl_ili9341_spi.c:33] lvgl_lcd_init(): lvgl_lcd_init
[  3308][D][lvgl_ili9341_spi.c:75] lvgl_lcd_init(): Initialization with vendor config
[  3313][D][lvgl_ili9341_spi.c:76] lvgl_lcd_init(): Vendor config value (ili9341_vendor_config_t[]){{.init_cmds=(ili9341_lcd_init_cmd_t[]){{.cmd=0xCF,.data=(uint8_t[]){0x00,0xC1,0x30},.data_bytes=3},{.cmd=0xED,.data=(uint8_t[]){0x64,0x03,0x12,0x81},.data_bytes=4},{.cmd=0xE8,.data=(uint8_t[]){0x85,0x00,0x78},.data_bytes=3},{.cmd=0xCB,.data=(uint8_t[]){0x39,0x2C,0x00,0x34,0x02},.data_bytes=5},{.cmd=0xF7,.data=(uint8_t[]){0x20},.data_bytes=1},{.cmd=0xEA,.data=(uint8_t[]){0x00,0x00},.data_bytes=2},{.cmd=0xC0,.data=(uint8_t[]){0x10},.data_bytes=1},{.cmd=0xC1,.data=(uint8_t[]){0x00},.data_bytes=1},{.cmd=0xC5,.data=(uint8_t[]){0x30,0x30},.data_bytes=2,},{.cmd=0xC7,.data=(uint8_t[]){0xB7},.data_bytes=1},{.cmd=0x3A,.data=(uint8_t[]){0x55},.data_bytes=1},{.cmd=0x36,.data=(uint8_t[]){0x08},.data_bytes=1},{.cmd=0xB1,.data=(uint8_t[]){0x00,0x1a},.data_bytes=2},{.cmd=0xB6,.data=(uint8_t[]){0x08,0x82,0x27},.data_bytes=3},{.cmd=0xF2,.data=(uint8_t[]){0x00},.data_bytes=1},{.cmd=0x26,.data=(uint8_t[]){0x01},.data_bytes=1},{.cmd=0xE0,.data=(uint8_t[]){0x0F,0x2A,0x28,0x08,0x0E,0x08,0x54,0xA9,0x43,0x0A,0x0F,0x00,0x00,0x00,0x00},.data_bytes=15},{.cmd=0xE1,.data=(uint8_t[]){0x00,0x15,0x17,0x07,0x11,0x06,0x2B,0x56,0x3C,0x05,0x10,0x0F,0x3F,0x3F,0x0F},.data_bytes=15},{.cmd=0x2B,.data=(uint8_t[]){0x00,0x00,0x01,0x3F},.data_bytes=4},{.cmd=0x2A,.data=(uint8_t[]){0x00,0x00,0x00,0xEF},.data_bytes=4},{.cmd=0x21},{.cmd=0x11,.delay_ms=120},{.cmd=0x29,.delay_ms=1}},.init_cmds_size=23}}
[  3918][D][esp32_smartdisplay.c:109] smartdisplay_lcd_set_backlight(): smartdisplay_lcd_set_backlight. duty:0.500000
[  3919][D][lvgl_xpt2046_spi.c:34] lvgl_touch_init(): lvgl_touch_init
[  3927][D][esp32_smartdisplay.c:36] lvgl_update_callback(): lvgl_update_callback

Screen always white.
Thank you

**Comment by MikolajPieklo** [comment]
Hello, 
On the 4848S040CIY3 board, demo doesn't work as well. The screen is black, but the touch is working. 

ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x1 (POWERON),boot:0x18 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0x44c
load:0x403c9700,len:0xbe4
load:0x403cc700,len:0x2a38
entry 0x403c98d4
E (125) esp_core_dump_flash: Incorrect size of core dump image: -16776961
[   127][I][esp32-hal-psram.c:96] psramInit(): PSRAM enabled
[   382][I][main.cpp:25] setup(): Board: esp32-4848S040CIY3
[   383][I][main.cpp:26] setup(): CPU: ESP32-S3 rev0, CPU Freq: 240 Mhz, 2 core(s)
[   384][I][main.cpp:27] setup(): Free heap: 314864 bytes
[   389][I][main.cpp:28] setup(): Free PSRAM: 8386295 bytes
[   394][I][main.cpp:29] setup(): SDK version: v4.4.4
[   399][D][esp32_smartdisplay.c:209] smartdisplay_init(): smartdisplay_init
[   406][D][lvgl_st7701_par.c:25] lvgl_lcd_init(): lvgl_lcd_init
[   676][D][esp32_smartdisplay.c:109] smartdisplay_lcd_set_backlight(): smartdisplay_lcd_set_backlight. duty:0.500000
[   676][D][lvgl_gt911_i2c.c:167] lvgl_touch_init(): lvgl_touch_init
[   682][D][lvgl_gt911_i2c.c:179] lvgl_touch_init(): i2c_param_config. host: 0
[   689][D][lvgl_gt911_i2c.c:181] lvgl_touch_init(): i2c_driver_install host: 0
[   695][D][lvgl_gt911_i2c.c:197] lvgl_touch_init(): esp_lcd_new_panel_io_i2c. host: 0
[   704][D][lvgl_gt911_i2c.c:201] lvgl_touch_init(): GT911 productId: 911
[   710][D][lvgl_gt911_i2c.c:202] lvgl_touch_init(): GT911 fwId: 1060
[   716][D][lvgl_gt911_i2c.c:203] lvgl_touch_init(): GT911 xResolution/yResolution: (480, 480)
[   724][D][lvgl_gt911_i2c.c:204] lvgl_touch_init(): GT911 vendorId: 00
[   731][D][lvgl_gt911_i2c.c:248] lvgl_touch_init(): esp_lcd_touch_new_i2c_gt911. host: 0
[   738][D][esp32_smartdisplay.c:36] lvgl_update_callback(): lvgl_update_callback
[ 20340][D][lvgl_gt911_i2c.c:148] gt911_lvgl_touch_cb(): Pressed at: (137,268), strength: 1070512958
[ 20370][D][lvgl_gt911_i2c.c:148] gt911_lvgl_touch_cb(): Pressed at: (137,268), strength: 1070512958

**Comment by rzeldent** [comment]
Hi,

Just got my new CYD board esp32-2432S028R with USB-C. This is a different board that I had and developed the software for. Was on sale on AliExpress.
Can confirm that the display remains white and no "demo". Both esp32-2432S028R and esp32-2432S028Rv2. Looks like a new variant... Will investigate this later.

Regarding the esp32-4848S040CIY3. Retested with my board (esp32-4848S040CIY1) and is working perfectly.

**Reply by MikolajPieklo** [reply]
Do you know whether 4848S040CIY1 and 4848S040CIY3 use the same pins connected to the display? Unfortunately, I can't find the PCB schematic for these boards...

**Reply by rzeldent** [reply]
Yes they are. Difference is the # relays, 1 or 3.
I informed for the schematic but to no avail. Has been reverse engineered from the demo.
A schema would be handy. Have no clue the pushbutton on the side is connected to.
See develop repo for high res image!

**Comment by chrbcd** [comment]
Hello,
I found the card box and saw the word 7789.
![image](https://github.com/rzeldent/esp32-smartdisplay/assets/44877185/f655054d-7dcf-45b9-b62d-78897f678560)
which translates to : 
2.8 TFT yellow board
240(RGB) *320
with touch 7789
Batch: 40105
 
I duplicated the json from the ESP32-2408-S8R to ESP32-2408-S8R7789 and changed all references from ILI9341 to ST7789. I also changed the following values:
ST7789_SPI_BUS_MISO_IO_NUM=GPIO_NUM_NC
ST7789_SPI_CONFIG_SPI_MODE=SPI_MODE3
LCD_MIRROR_X=false
(I checked, we can't read the registers so no MISO)
and now everything works correctly.

The aliexpress link is as follows
https://www.aliexpress.us/item/3256805924045422.html?spm=a2g0o.order_list.order_list_main.33.37925e5bhhZViL&gatewayAdapt=glo2usa4itemAdapt

![image](https://github.com/rzeldent/esp32-smartdisplay/assets/44877185/c439987d-1f0c-4d64-bb8b-f2f1c4ab08fe)

Where are we going if the Chinese copy each other?

Sorry for the trouble generated, and thank you for your wonderful development.

**Reply by rzeldent** [reply]
Hi Christian,

Thanks for your post. I think this was exactly what a lot of other users (and I) were waiting for. Now, looking at the box I also see that 7789 is on the label but did not noticed it.

I made a new definition: esp32-2432S028Rv3 that does work on this board! This board can be noticed easily as it has an USB-C and micro USB! 

Yes, there are a lot of clones out there! Difficult to keep up with implementing the definitions but trying!

Thanks again and kudos!

**Reply by rzeldent** [reply]
Just committed to the [develop ](https://github.com/rzeldent/esp32-smartdisplay-demo/tree/develop) branch

**Comment by zdid** [comment]
Hello,
I tested with esp32-2432s028-v3 and esp32-smartdisplay 2.1.
In portrait mode: ok
in landscape mode (lv_disp_set_rotation(disp, LV_DISP_ROT_90); ): the touch does not work correctly and it is in portrait mode.
I don't know how to use the develop branch
Please tell me what to do.

**Reply by MikolajPieklo** [reply]
Iif I remember correctly, and you want to change the branch in platformio, you should open platformio.ini and add "#develop" on the end

example: 
lib_deps = https://github.com/rzeldent/esp32-smartdisplay#develop

**Reply by zdid** [reply]
Ok thanks for the develop branch. I use.
but the touch does not work correctly in landscape mode.  It's quite weird!

**Reply by zdid** [reply]
to complete: 
x= 0 and y 0 are in the upper right corner (relative to the display).
at the top left of the screen, I have the message
[Warn] (482.411, +30) indev_pointer_proc: Y is 290 which is greater than ver. res (in lv_indev.c line #368)
Is this enough information?

**Reply by zdid** [reply]
It s OK with     
-D TOUCH_SWAP_XY=true
-D TOUCH_SWAP_X=true
-D TOUCH_SWAP_Y=false

Thanks for esp32-smartdisplay!!!
